src/ex1/script.js

```diff
@@ -0,0 +1,141 @@
+        </svg>
+    </button>
+    `;
+  taskId++;
+  return task;
+};
+
+const onAddTask = (event) => {
+  // Validate the input isn't empty before adding the task
+  if (!addTaskInput.value) {
+    return;
+  }
+
+  document.querySelector(".tasks-container").appendChild(createTask());
+
+  // Clear 'Add task' input
+  addTaskInput.value = "";
+  addTaskBtn.classList.remove("valid-task-btn");
+  isAddTaskClickable = false;
+
+  tasksCnt++;
+  updateTasksLeft();
+
+  if (tasksCnt === 1) {
+    // Hide finished all
+    document
+      .querySelector(".finished-all-missions")
+      .classList.toggle("finished-all-missions-active");
+    // Show clear all and left tasks
+    document.querySelector(".todo-footer-container").classList.toggle("hide");
+  }
+
+  const checkboxBtn = document.querySelector(`#input-${taskId - 1}`);
+  const taskTxt = document.querySelector(`#task-txt-${taskId - 1}`);
+  checkboxBtn.addEventListener("click", (event) => onChecked(event, taskTxt));
+
+  const input = document.querySelector(`#task-${taskId - 1}`);
+  input.addEventListener("click", (event) => onTaskPress(event));
+
+  const trashBtn = document.querySelector(`#trash-button-${taskId - 1}`);
+  trashBtn.addEventListener("click", (event) => onTaskDelete(event));
+};
+
+const onChecked = (event, taskTxt) => {
+  taskTxt.classList.toggle("done-task-txt");
+};
+
+const onTaskPress = (event) => {
+  if (event.target.classList.length) {
+    alert(document.querySelector(`#task-txt-${taskId - 1}`).innerText);
+  }
+};
+
+// Delete task
+const onTaskDelete = (event) => {
+  const clickedId = event.target.id;
+  const taskIdToRemove = clickedId.slice(clickedId.lastIndexOf("-") + 1);
+  document.querySelector(`#task-${taskIdToRemove}`).remove();
+
+  tasksCnt--;
+  updateTasksLeft();
+
+  if (!tasksCnt) {
+    // Show finished all
+    document
+      .querySelector(".finished-all-missions")
+      .classList.toggle("finished-all-missions-active");
+    // Hide clear all and left tasks
+    document.querySelector(".todo-footer-container").classList.toggle("hide");
+  }
+};
+
+// Clear all
+const clearAllBtn = document.querySelector(".clear-all");
+clearAllBtn.addEventListener("click", (event) => onClearAll(event));
+
+const onClearAll = (event) => {
+  const nodes = document.querySelector(".tasks-container").childNodes;
+  console.log(nodes);
+  // The first 5 aren't tasks
+  for (let i = nodes.length - 1; i >= 5; i--) {
```

Comment:
what are the first 5 elements? seems like a buggy thingy.
you shouldn't have non-tasks elements when pulling tasks via this querySelector, this means your not using the class name proprely.

Reply:
The first 5 are:
1. text
2. Add task div
3. text
4. finished all tasks div
5. text

Deffenetly agree with you. Wanted to change it in Ex1 but didn't have enough time. Will be refactored in Ex2.